zanata-war/src/main/java/org/zanata/security/SecurityFunctions.java

```diff
@@ -66,29 +64,34 @@
  *         href="mailto:camunoz@redhat.com">camunoz@redhat.com</a>
  */
 @Slf4j
-public class SecurityFunctions {
-    protected SecurityFunctions() {
-    }
+public class SecurityFunctions extends PermissionProvider {
+
+    @Inject
+    private ZanataIdentity identity;
+
+    @Inject
+    @Authenticated
+    private HAccount authenticatedAccount;
 
     /* admin can do anything */
     @GrantsPermission
-    public static boolean isAdmin() {
-        return getIdentity().hasRole("admin");
+    public boolean isAdmin() {
+        return identity.hasRole("admin");
     }
 
-    public static boolean isProjectMaintainer(HProject project) {
+    public boolean isProjectMaintainer(HProject project) {
         return isProjectRole(project, Maintainer);
     }
 
-    public static boolean isProjectTranslationMaintainer(HProject project) {
+    public boolean isProjectTranslationMaintainer(HProject project) {
         return isProjectRole(project, TranslationMaintainer);
     }
 
     /* Check whether the authenticated person has the given role in the given
      * project.
      */
-    private static boolean isProjectRole(HProject project, ProjectRole role) {
-        Optional<HAccount> account = getAuthenticatedAccount();
+    private boolean isProjectRole(HProject project, ProjectRole role) {
+        Optional<HAccount> account = Optional.fromNullable(authenticatedAccount);
```

Comment:
The fact that it's `@Dependent` should help with this, I think.

On 15 February 2016 at 11:49, Carlos Munoz <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/security/SecurityFunctions.java
> <https://github.com/zanata/zanata-server/pull/1093#discussion_r52853814>:
>
> >          return isProjectRole(project, TranslationMaintainer);
> >      }
> >
> >      /* Check whether the authenticated person has the given role in the given
> >       * project.
> >       */
> > -    private static boolean isProjectRole(HProject project, ProjectRole role) {
> > -        Optional<HAccount> account = getAuthenticatedAccount();
> > +    private boolean isProjectRole(HProject project, ProjectRole role) {
> > +        Optional<HAccount> account = Optional.fromNullable(authenticatedAccount);
>
> It is possible. However, it think that the producer would be for @Authenticated
> Optional<?> objects. I can simply add it right besides the @Authenticated
> HAccount producer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1093/files#r52853814>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat